src/main/java/com/qunite/api/service/QueueServiceImpl.java

```diff
@@ -14,19 +14,30 @@
 import java.util.List;
 import java.util.Optional;
 import java.util.function.BiConsumer;
-import lombok.RequiredArgsConstructor;
 import one.util.streamex.StreamEx;
+import org.springframework.context.annotation.Lazy;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
 import org.springframework.transaction.annotation.Transactional;
 
-@RequiredArgsConstructor
 @Service
 public class QueueServiceImpl implements QueueService {
   private final QueueRepository queueRepository;
   private final UserRepository userRepository;
   private final EntryRepository entryRepository;
   private final UserService userService;
+  private final QueueServiceImpl self;
+
+  public QueueServiceImpl(QueueRepository queueRepository, UserRepository userRepository,
+                          EntryRepository entryRepository, UserService userService,
+                          @Lazy QueueServiceImpl self) {
```

Comment:
https://stackoverflow.com/questions/59505213/how-to-use-lazy-annotation-in-a-class-constructor-with-lombok